yt/yt/server/queue_agent/queue_controller.cpp

```diff
@@ -510,66 +508,101 @@ class TOrderedDynamicTableController
 
             ProfileManager_->Profile(previousQueueSnapshot, nextQueueSnapshot);
 
+            UpdateExports(nextQueueSnapshot);
+
             if (ShouldTrim(nextQueueSnapshot->PassIndex)) {
                 Trim();
             }
+        }
 
-            const auto& staticExportConfig = nextQueueSnapshot->Row.StaticExportConfig;
+        YT_LOG_INFO("Queue controller pass finished");
+    }
+
+    void UpdateExports(const TQueueSnapshotPtr& nextQueueSnapshot)
+    {
+        auto enableQueueStaticExport = DynamicConfig_.Acquire()->EnableQueueStaticExport;
+        const auto& staticExportConfig = nextQueueSnapshot->Row.StaticExportConfig;
+
+        auto guard = WriterGuard(QueueExportsLock_);
+
+        if (staticExportConfig) {
+            if (auto staticExportConfigError = CheckStaticExportConfig(*staticExportConfig); !staticExportConfigError.IsOK()) {
+                QueueExports_ = staticExportConfigError;
+                return;
```

Comment:
I don't think we discussed this, this is probably an oversight.

Reply:
Wait, but after assignment old exports are destructed, and with them their executors, so that should be ok, I think. Or am I misunderstanding something?

Reply:
You are right, my bad. This setup with TErrorOr<something useful> as a field is a bit unusual and confusing at first.
But I don't see anything actually wrong with it, so let's keep it.